### PR TITLE
Accessible description

### PR DESCRIPTION
**Pull request by aarongustafson**
Updates per #7

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/image-resource/pull/28.html" title="Last updated on Feb 3, 2021, 12:14 AM UTC (6a7cf4c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/image-resource/28/d9f14d9...6a7cf4c.html" title="Last updated on Feb 3, 2021, 12:14 AM UTC (6a7cf4c)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -180,6 +187,31 @@ <h3>
...
+      <section>
+        <h3>
+          <dfn>label</dfn> member
+        </h3>
+        <p>
+          The {{ImageResource/label}} of an {{ImageResource}} is a string that
+          provides an accessible name for the associated image.
+        </p>
+        <p>
+          The {{ImageResource/label}} is strongly recommended, but optional as
```
Comment (marcoscaceres) [inline index.html:199]: Recommended already implies optional. 

```suggestion
          The {{ImageResource/label}} is strongly recommended as
```

File: index.html [hunk 2]
```diff
@@ -223,6 +255,22 @@ <h2>
...
         </li>
+        <li>If |input|'s {{ImageResource/label}} member is present and not the
+        empty string:
+          <ol>
+            <li>Set |image|'s [=image resource/label=] to
+            |input|.{{ImageResource/label}}.
+            </li>
+          </ol>
+        </li>
+        <li>If |input|'s {{ImageResource/label}} member is not present and an
```
Comment (marcoscaceres) [inline index.html:266]: Wait, doesn't substitution assume the resource is fetched? If so, that probably shouldn't be part of this algorithm (same below).

File: index.html (hunk 1 above)
Comment (LJWatson) [reply index.html:199]: +1 to @marcoscaceres' proposed change.

File: index.html (hunk 2 above)
Comment (aarongustafson) [reply index.html:266]: That makes sense, substitution only needs to happen if that image resource is selected for use. Where would you capture that? It's not part of the parsing algorithm, so it's party of the… ?

**Comment by aarongustafson** [comment]
Todo: fix the external label references

File: index.html (hunk 2 above)
Comment (marcoscaceres) [reply index.html:266]: This goes back to this being somewhat wishful thinking/false hope... it's not something implementations will do (even with SVGs, they will be treated as static images). I don't think we should mention it at all to be honest. The right thing to do is to add the labels or let other parts (e.g., short_name) do the right thing.

**Comment by aarongustafson** [comment]
> Assuming that this will be exposed to the platform accessibility API in the same way an HTML img/alt is, shouldn't this refer to "accessible name" not "accessible description? Or is the description in this case ancillary to something else that provides the accessible name?

Yes. That has been resolved now. Changed the string to "label" as well (to avoid confusion).

File: index.html [hunk 3]
```diff
@@ -198,10 +198,11 @@ <h3>
           associated image.
         </p>
         <p>
-          You MUST provide a {{ImageResource/label}} unless the {{ImageResource}}'s label can be inferred from
-          an <dfn>external label</dfn> in its context. For example, the
-          {{WebAppManifest/name}} can act as the {{ImageResource/label}} for
-          any member of the {{WebAppManifest/icons}} array.
+          You MUST provide a {{ImageResource/label}} unless the
```
Comment (marcoscaceres) [inline index.html:199]: The "You MUST" here is kinda problematic... RFC2119 keywords only apply to conformance classes, not human authors.  
```suggestion
          Authors are strongly encouraged to add a {{ImageResource/label}} unless the
```

File: index.html [hunk 4]
```diff
@@ -103,20 +103,27 @@ <h2>
         <dd>
           An [=image MIME type=].
         </dd>
+        <dt>
+          <dfn data-dfn-for="image resource">label</dfn>
+        </dt>
+        <dd>
+          A [=string=] representing the [[[#dfn-accessible-name]]] of the image.
```
Comment (marcoscaceres) [inline index.html:110]: I think we want: 

```suggestion
          A [=string=] representing the [=accessible name=] of the image.
```

And then we need to get the ACCNAME spec to export that, right?

File: index.html [hunk 5]
```diff
@@ -103,20 +103,27 @@ <h2>
...
+        </dd>
       </dl>
       <p>
         In an API, this is represented as an {{ImageResource}} dictionary.
       </p>
     </section>
-    <section data-dfn-for="ImageResource">
+    <section data-dfn-for="ImageResource" data-link-for="ImageResource">
```
Comment (marcoscaceres) [inline index.html:117]:
```suggestion
    <section data-dfn-for="ImageResource">
```

File: index.html [hunk 6]
```diff
@@ -103,20 +103,27 @@ <h2>
...
       <h2>
-        <dfn>ImageResource</dfn> dictionary
+        <dfn data-dfn-for="">ImageResource</dfn> dictionary
```
Comment (marcoscaceres) [inline index.html:119]:
```suggestion
        <dfn>ImageResource</dfn> dictionary
```

File: index.html [hunk 7]
```diff
@@ -180,6 +187,31 @@ <h3>
...
+          You MUST provide a {{ImageResource/label}} unless the
+          {{ImageResource}}'s label can be inferred from an <dfn>external
+          label</dfn> in its context. For example, the {{WebAppManifest/name}}
+          can act as the {{ImageResource/label}} for any member of the
+          {{WebAppManifest/icons}} array.
+        </p>
+        <p>
+          If an explicit {{ImageResource/label}} is not provided, but the
+          {{ImageResource/type}} supports providing an [[[#dfn-accessible-name]]],
+          implementors may choose to infer the {{ImageResource/label}} until
```
Comment (marcoscaceres) [inline index.html:208]:
```suggestion
          implementors MAY choose to infer the {{ImageResource/label}} until
```

File: index.html [hunk 8]
```diff
@@ -268,6 +316,22 @@ <h2>
...
+            <li>Set |image|'s [=image resource/label=] to the value of
+            |input|'s "label" property.
+            </li>
+          </ol>
+        </li>
+        <li>If |input|'s {{ImageResource/label}} member is not present and an
+        |=external label=| can be substituted:
+          <ol>
+            <li>Set |image|'s [=image resource/label=] to the value of
+            |=external label=|.
```
Comment (marcoscaceres) [inline index.html:331]:
```suggestion
            [=external label=].
```

File: index.html (hunk 4 above)
Comment (aarongustafson) [reply index.html:110]: The exporting is where I was running into issues. How do we manage that?

Comment (aarongustafson) [reply index.html]: I think this is it: https://github.com/w3c/respec/wiki/data-export

**Comment by aarongustafson** [comment]
@marcoscaceres @rayankans Any blockers to this merging?

**Comment by LJWatson** [comment]
The PR LGTM but it looks like there is a check that is failing.

**Review by rayankans — APPROVED** [review]

**Comment by aarongustafson** [comment]
All ReSpec issues have been resolved. Merging now.

**Pull request merged**